app/models/node.rb

```diff
@@ -390,8 +390,10 @@ def response_count(key = 'response')
   end
 
   # power tags have "key:value" format, and should be searched with a "key:*" wildcard
-  def has_power_tag(key)
-    !power_tag(key).blank?
+  def has_power_tag(key, preview_tags = [])
+    return !power_tag(key).blank? if preview_tags.blank?
+    preview_tag = preview_tags.join(" ").match(key)
+    !preview_tag.blank?
```

Comment:
We are using these 2 lines again in line 482,483...can we make a small util function and call it at both places?